hphp/runtime/ext/xenon/ext_xenon.cpp

```diff
@@ -193,10 +201,12 @@ void Xenon::start(uint64_t msec) {
 
     sigevent sev={};
     sev.sigev_notify = SIGEV_SIGNAL;
-    sev.sigev_signo = SIGVTALRM;
-    sev.sigev_value.sival_ptr = nullptr; // null for Xenon signals
+    sev.sigev_signo = SIGPROF;
     timer_create(CLOCK_REALTIME, &sev, &m_timerid);
 
+    signal(SIGPROF, SIG_IGN);
```

Comment:
One more small comment is that `signal` should probably go above `timer_create` to avoid these kinds of circumstances:
- `hhvm` receives a `SIGSTP`/`SIGSTOP` after `timer_create` and is resumed later with `SIGCONT`, potentially causing the realtime timer to trigger immediately before `SIGPROF` is ignored
- Some long time window where other threads execute between `timer_create` and `signal`

Reply:
My understanding is that the timer isn't really running before `timer_settime()`.  If so, there is no need to move it above.

Reply:
You're completely right, I got mixed up. Sorry for the confusion!